astroquery/heasarc/tests/parametrization.py

```diff
@@ -43,22 +46,63 @@ def filename_for_request(url, params, output=False):
     return data_path(fileid, output=output)
 
 
+def request_dict_to_string_list(request_dict):
+    return json.dumps(request_dict, sort_keys=True, indent=4).split("\n")
+
+
 def get_mockreturn(session, method, url, params=None, timeout=10, **kwargs):
 
     filename = filename_for_request(url, params)
     try:
-        content = open(filename, "rt").read()
+        with open(filename, "rt") as f:
+            content = json.load(f)['response_text']    
+    except (json.decoder.JSONDecodeError, IndexError) as e:
+        log.error(
+            f'stored mock data in {filename} for url="{url}" and params="{params}"',
+            f'causes IndexError: {e}. You might need to regenerate it.'
+        )
+        raise
     except FileNotFoundError:
         log.error(
             f'no stored mock data in {filename} for url="{url}" and params="{params}", '
             f'perhaps you need to clean test data and regenerate it? '
             f'It will be regenerated automatically if cleaned, try `rm -fv astroquery/heasarc/tests/data/* ./build`'
         )
+
+        unavailable_request = request_to_dict(url, params)
+
+        log.error('overall, have %s mock data entries', len(list_mock_data()))
+
+        
+                
+        available_requests = []
+        for available_mock_data in list_mock_data():
+            with open(available_mock_data) as f:
+                available_request = json.load(f)['request']
+                a = request_dict_to_string_list(available_request)
+                ua = request_dict_to_string_list(unavailable_request)
+                try:
+                    available_requests.append([
+                        SequenceMatcher(a=a, b=ua).ratio(),
+                        Differ().compare(a=a, b=ua),
+                    ])
```

Comment:
could you add a little commentary about what this does?  I'm not familiar with the difflib library.  This seems like something that could be more generally useful if I understood it better.

Reply:
It produces a diff between the request which was we are trying to substitute with mock response, and available mock requests. Sorting by similarity.
The diff looks like the on `pytest` uses to detail exceptions in `assert`.
I will add an example later, when I have time to work on this (not in the next couple of days).